crates/bevy_ecs/src/system/query.rs

```diff
@@ -1357,12 +1357,59 @@ impl<'w, 's, Q: WorldQuery, F: ReadOnlyWorldQuery> IntoIterator for &'w mut Quer
     }
 }
 
-/// An error that occurs when retrieving a specific [`Entity`]'s component from a [`Query`]
+/// An error that occurs when retrieving a specific [`Entity`]'s component from a [`Query`].
 #[derive(Debug, PartialEq, Eq)]
 pub enum QueryComponentError {
+    /// The [`Query`] does not have read access to the requested component.
+    /// 
+    /// # Example
+    ///
+    /// ```
+    /// # use bevy_ecs::{prelude::*, system::QueryComponentError};
+    /// #
+    /// # #[derive(Component)]
+    /// # struct OtherComponent;
+    /// #
+    /// # #[derive(Component)]
+    /// # struct RequestedComponent;
+    /// #
+    /// # #[derive(Resource)]
+    /// # struct SpecificEntity {
+    /// #     entity: Entity,
+    /// # }
+    /// #
+    /// fn get_missing_read_access_error(query: Query<&OtherComponent>, res: Res<SpecificEntity>) {
+    ///     if let Err(QueryComponentError::MissingReadAccess) = query.get_component::<RequestedComponent>(res.entity) {
+    ///         println!("query doesn't have read access to RequestedComponent");
+    ///     }
+    /// }
+    /// # bevy_ecs::system::assert_is_system(get_missing_read_access_error);
```

Comment:
Is having an example for an error variant common? Not that I have a problem with this.

Reply:
I thought it would be better to have an example since it's more straightforward to explain how it might happen, but I don't know honestly.

Reply:
I am fine with merging this as is, but I'd like to defer this question to others and see what they think.

Reply:
I really like the doc test here: I think it's quite clear.